lib/Dialect/TritonGPU/Transforms/Pipeliner/TMAStoresPipeline.cpp

```diff
@@ -101,6 +109,9 @@ bool mlir::triton::pipelineTMAStores(scf::ForOp forOp) {
     storeToAlloc[store.op] = alloc;
   }
 
+  bool hasDeviceSideTMA = llvm::any_of(tmaStores, [](const TMAStore &store) {
+    return !triton::isHostSideDescriptor(store.desc);
+  });
```

Comment:
Can you check for the presence of `MakeTensorDescOp` inside the loop? That would more-directly test what you're looking for.

Reply:
ah right, I guess in the previous logic it wasn't the same but now it is indeed

Reply:
actually I take that back. The point is to not apply it to if the descriptor are not related to the TMA store.
This is not much better but the check seems cheap enough